### PR TITLE
Adding component for installing Caddy

### PR DESCRIPTION
**Pull request by blarghmatey**
Caddy is the web server that we are using for new installations due to its ease of deployment and flexible usage patterns.

### CODE REVIEW

File: src/bilder/components/caddy/packer.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  business_unit = "operations"
+  app_name = "concourse"
+}
+
+variable "build_environment" {
+  type = string
+  default = "operations-qa"
+}
+
+# Available options are "web" or "worker". Used to determine which type of node to build an image for.
```
Comment (shaidar) [inline src/bilder/components/caddy/packer.pkr.hcl:12]: Why is this needed here?

File: src/bilder/components/caddy/packer.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    inline = [
+      "echo ${build.name}",
+      "echo ${build.ID}",
+      "echo ${build.ConnType}",
+      "echo '${build.SSHPrivateKey}' > /tmp/packer-session.pem",
+      "chmod 600 /tmp/packer-session.pem"
+    ]
+  }
+  provisioner "shell-local" {
+    except = ["docker.concourse"]
```
Comment (shaidar) [inline src/bilder/components/caddy/packer.pkr.hcl]: Why this exception?

File: src/bilder/components/caddy/packer.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  }
+  snapshot_tags = {
+    OU      = "${local.business_unit}"
+    app     = "${local.app_name}"
+    purpose = "${local.app_name}-${var.node_type}"
+  }
+  # Base all builds off of the most recent Debian 10 image built by the Debian organization.
+  source_ami_filter {
+    filters = {
+      name                = "debian-10-amd64*"
```
Comment (shaidar) [inline src/bilder/components/caddy/packer.pkr.hcl:31]: Should this be defined in the model instead?

File: src/bilder/components/caddy/steps.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            ensure_home=False,
+            state=state,
+            host=host,
+        )
+        files.download(
+            name="Download custom build of Caddy",
+            dest="/usr/local/bin/caddy",
+            src=caddy_config.custom_download_url(),
+            mode=755,
+            sttate=state,
```
Comment (shaidar) [inline src/bilder/components/caddy/steps.py]: Typo

File: src/bilder/components/caddy/packer.pkr.hcl (hunk 2 above)
Comment (blarghmatey) [reply src/bilder/components/caddy/packer.pkr.hcl]: Copy :spaghetti:

File: src/bilder/components/caddy/packer.pkr.hcl (hunk 3 above)
Comment (blarghmatey) [reply src/bilder/components/caddy/packer.pkr.hcl:31]: Ideally the config should be agnostic to what it's actually being launched on so I don't think it makes sense to mix those concerns.

File: src/bilder/components/caddy/packer.pkr.hcl (hunk 2 above)
Comment (shaidar) [reply src/bilder/components/caddy/packer.pkr.hcl]: I thought this was copy 🍝

**Comment by shaidar** [comment]
👍
